### PR TITLE
feat: support code component

### LINKED ISSUE #1771: [Feature] New component Code
# Feature request 🚀

### Expected:

- Add a new component Code to display code blocks and highlight code syntax
- No breaking changes

### PR DESCRIPTION
**Pull request by cszhjh**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

Describe your modifications here.

## Issues

close #1771

## Related Links

Links related to this pr.

### CODE REVIEW

File: packages/varlet-ui/src/code/Code.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <div :class="classes(n())">
```
Comment (chouchouji) [inline packages/varlet-ui/src/code/Code.vue:2]: If there is only one class, `classes` function is not necessary. `:class="n()"` is better.

File: packages/varlet-ui/src/code/Code.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      n,
+      classes,
+    }
+  },
+})
+</script>
+
+<style lang="less">
+@import '../styles/common';
+@import './code.less';
```
Comment (chouchouji) [inline packages/varlet-ui/src/code/Code.vue:40]: `./code.less` -> `./code`

File: packages/varlet-ui/src/code/docs/zh-CN.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+查看所有支持的[主题](https://shiki.style/themes)
+
+### CodeLanguage
+
+查看所有支持的[语言](https://shiki.style/languages)
+
+### 样式变量
+以下为组件使用的 css 变量，可以使用 [StyleProvider 组件](#/zh-CN/style-provider) 进行样式定制。
+
+#### Button Variables
```
Comment (chouchouji) [inline packages/varlet-ui/src/code/docs/zh-CN.md:50]: delete it

File: packages/varlet-ui/src/code/docs/en-US.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+### Intro
+
+Code component, used to display code blocks and highlight code syntax.
+
+### Notes
+
+Due to package size considerations, Varlet does not include a built-in code highlighter. If you need to use the code block component, please ensure you use the [HighlighterProvider component](#/en-US/highlighter-provider) to customize the highlighter.
+
+### Toggle Language
```
Comment (haoziqaq) [inline packages/varlet-ui/src/code/docs/en-US.md:11]: This block is recommended for removal. as switching languages or switching theme ​​is an uncommon use case.

File: packages/varlet-ui/src/highlighter-provider/props.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export interface Highlighter {
+  codeToHtml: (code: string, options: CodeToHtmlOptions) => Promise<string>
+}
+
+export const props = {
+  highlighter: {
+    type: Object as PropType<Highlighter>,
+    default: null,
```
Comment (haoziqaq) [inline packages/varlet-ui/src/highlighter-provider/props.ts:15]: remove `default: null`, highlighter defaults undefined

**Review by haoziqaq — COMMENTED** [review]
The documentation of `provider` and `code` should be merge into `code`.

File: packages/varlet-ui/src/code/docs/en-US.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+```html
+<script setup>
+  import { ref } from 'vue'
+  import { codeToHtml } from 'shiki'
+
```
Comment (haoziqaq) [inline packages/varlet-ui/src/code/docs/en-US.md:17]: These indents should be removed

**Comment by cszhjh** [comment]
> The documentation of `provider` and `code` should be merge into `code`.

done

File: packages/varlet-ui/src/code/docs/en-US.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+      const highlightedCode = hljs.highlight(code, { language: lang }).value
+      return `
+        <div class="${theme}">
+          <pre class="hljs"><code>${highlightedCode}</code></pre>
+        </div>
+      `
+    },
+  }
+}
```
Comment (haoziqaq) [inline packages/varlet-ui/src/code/docs/en-US.md:68]: Remove highlightjs supported

File: packages/varlet-ui/src/code/code.less [hunk 8]
```diff
@@ -0,0 +1,27 @@
+:root {
+  --code-margin: 16px 0;
```
Comment (haoziqaq) [inline packages/varlet-ui/src/code/code.less:2]: remove it.

File: packages/varlet-ui/src/code/Code.vue [hunk 9]
```diff
@@ -0,0 +1,45 @@
+<template>
+  <div :class="n()">
+    <div :class="n('content')" v-html="highlightedCode"></div>
```
Comment (haoziqaq) [inline packages/varlet-ui/src/code/Code.vue:3]: Whether it is possible to remove content blocks ?

Comment (cszhjh) [reply packages/varlet-ui/src/code/Code.vue:3]: of course

**Pull request merged**